### ISSUE TITLE
librealsense2 for armhf

### ISSUE THREAD

**Issue by ICH88** [bytedeco/javacpp-presets#945]
Hi, I'm a Java noob so please don't hate me for stupid questions. ;)

This isn't really an issue but more a question:
How can I get librealsense2 with JavaCV working on the armhf platform?

Are there any precompiled librealsense2 libraries for armhf?

**Comment by ICH88** [comment]
I compiled it following those instructions: https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments

I got 2 files:
librealsense2.jar
librealsense2-linux-armhf.jar

But no success: 
java.lang.UnsatisfiedLinkError: Platform "linux-armhf" not supported by class org.bytedeco.librealsense2.global.realsense2

**Comment by saudet** [comment]
We'll first need to add a build for linux-armhf to the cppbuild.sh build script here:
https://github.com/bytedeco/javacpp-presets/blob/master/libfreenect2/cppbuild.sh#L47

We can check what it looks like for libfreenect and do something like that:
https://github.com/bytedeco/javacpp-presets/blob/master/libfreenect/cppbuild.sh#L55

/cc @cansik @vb216

**Comment by ICH88** [comment]
What else needs to be done?

So far I've added this to the cppbuild.sh:
`    linux-armhf)
        cd ../libusb-$LIBUSB_VERSION
        CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ CFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" CXXFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" CPPFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" ./configure --prefix=$INSTALL_PATH --disable-shared --with-pic --host=arm-linux-gnueabihf --disable-udev
		make -j $MAKEJ
        make install
        cd ../librealsense-$LIBREALSENSE2_VERSION
        CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ CFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" CXXFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" CPPFLAGS="-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard" $CMAKE -DCMAKE_INSTALL_PREFIX="$INSTALL_PATH" -DLIBUSB_INC=$INSTALL_PATH/include/libusb-1.0/ -DLIBUSB_LIB=$INSTALL_PATH/lib/libusb-1.0.a -DBUILD_UNIT_TESTS=OFF -DBUILD_EXAMPLES=OFF -DBUILD_GRAPHICAL_EXAMPLES=OFF .
        make -j $MAKEJ
		make install
        ;;`

And I've added this line to the pom.xml under the armhf profile:
`<module>librealsense2</module>`

**Comment by saudet** [comment]
Something like that yeah. Please open a pull request and let's take a look at that. Thanks!

**Comment by ICH88** [comment]
So that took me a while to figure out, as I'm not familiar with java or maven ... but I got it to compile now!

I also had to edit the /javacpp-presets/librealsense2/src/main/java/org/bytedeco/librealsense2/presets/realsense2.java

I'll see if I can make a pull request soon as I'm running out of time right now.

Next step would be to get arm64 running, as I would also need that. ;)
I already started working on that.
